src/kOS.Safe/Execution/CPU.cs

```diff
@@ -413,11 +415,13 @@ public void AssertValidDelegateCall(IUserDelegate userDelegate)
         {
             if (userDelegate.ProgContext != currentContext)
             {
-                throw new KOSInvalidDelegateContextException(
-                    (currentContext == contexts[0] ? "the interpreter" : "a program"),
-                    (currentContext == contexts[0] ? "a program" : "the interpreter")
-                    );
-            }
+                string currentContextName = (currentContext == contexts[0] ? "the interpreter" : "a program");
+                string delegateContextName = (
+                    userDelegate.ProgContext == contexts[0] ? currentContextName = "the interpreter" : (
+                        currentContext == contexts[0] ? "a program" : "a different program from a previous run" ) );
+
+                throw new KOSInvalidDelegateContextException(currentContextName, delegateContextName);
+           }
```

Comment:
For some reason I didn't see this line comment when you first made it. 

Go ahead and change it if you prefer.

I understand that it looks ugly, but it's not just a syntax difference from an``if`` block.  It's also a design pattern difference because essentially using ``?`` ``:`` is functional thinking while using ``if``/``else`` is procedural thinking.  I just wish it looked better.  It's algorithmically prettier, but syntactically uglier.